mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/ui/HistoryFragment.java

```diff
@@ -76,11 +96,83 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         mPresenter.attachView(this);
 
         setupUi();
+        setStatusList();
         mPresenter.fetchTransactions();
 
         return rootView;
     }
 
+    @Override
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        super.onCreateOptionsMenu(menu, inflater);
+        inflater.inflate(R.menu.menu_filter, menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.menu_filter_history:
+                showFilterDialog();
```

Comment:
A good idea is to check for `getActivity() != null` as the options menu is inflated as a part of activity and is independent of the fragment's UI. Fragment might have been detached but the options menu might still be there. Clicking in suich cases will result in crash. Can also check if fragment is still attached by using `isAdded()`